Dockerfile-alpine

```diff
@@ -0,0 +1,166 @@
+# References:
+# * https://github.com/felixbuenemann/vips-alpine
+# * https://hub.docker.com/r/frolvlad/alpine-glibc/dockerfile/
+# * https://github.com/sgerrand/alpine-pkg-glibc/releases
+# * http://kefblog.com/2017-07-04/Golang-ang-docker
+
+FROM golang:1.11.5-alpine3.9 AS build
+
+ENV BUILD_DEPS \
+    alpine-sdk \
+    build-base \
+    ca-certificates \
+    expat-dev \
+    fftw-dev \
+    gawk \
+    giflib-dev \
+    glib-dev \
+    gobject-introspection-dev \
+    gtk-doc \
+    imagemagick-dev \
+    lcms2-dev \
+    libexif-dev \
+    libgsf-dev \
+    libimagequant-dev \
+    libintl \
+    libjpeg-turbo-dev \
+    libpng-dev \
+    librsvg-dev \
+    libwebp-dev \
+    libxml2-dev \
+    openexr-dev \
+    orc-dev \
+    pango-dev \
+    poppler-dev \
+    py-gobject3-dev \
+    tiff-dev \
+    wget \
+    zlib-dev
+ENV VIPS_DIR /vips
+
+# Versions
+ARG VIPS_VERSION=8.7.4
+ARG IMAGINARY_VERSION=1.0.18
+ARG ALPINE_GLIBC_PACKAGE_VERSION=2.29-r0
+
+ENV PKG_CONFIG_PATH "${VIPS_DIR}/lib/pkgconfig:$PKG_CONFIG_PATH"
+ENV LANG C.UTF-8
+
+#-------------------------------------------------------------------------------
+
+# https://github.com/sgerrand/alpine-pkg-glibc/releases
+ARG ALPINE_GLIBC_BASE_URL="https://github.com/sgerrand/alpine-pkg-glibc/releases/download"
+ARG ALPINE_GLIBC_BASE_PACKAGE_FILENAME="glibc-$ALPINE_GLIBC_PACKAGE_VERSION.apk"
+ARG ALPINE_GLIBC_BIN_PACKAGE_FILENAME="glibc-bin-$ALPINE_GLIBC_PACKAGE_VERSION.apk"
+ARG ALPINE_GLIBC_I18N_PACKAGE_FILENAME="glibc-i18n-$ALPINE_GLIBC_PACKAGE_VERSION.apk"
+
+# Add public key for Alpine glibc packages
+RUN echo \
+    "-----BEGIN PUBLIC KEY-----\
+    MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEApZ2u1KJKUu/fW4A25y9m\
+    y70AGEa/J3Wi5ibNVGNn1gT1r0VfgeWd0pUybS4UmcHdiNzxJPgoWQhV2SSW1JYu\
+    tOqKZF5QSN6X937PTUpNBjUvLtTQ1ve1fp39uf/lEXPpFpOPL88LKnDBgbh7wkCp\
+    m2KzLVGChf83MS0ShL6G9EQIAUxLm99VpgRjwqTQ/KfzGtpke1wqws4au0Ab4qPY\
+    KXvMLSPLUp7cfulWvhmZSegr5AdhNw5KNizPqCJT8ZrGvgHypXyiFvvAH5YRtSsc\
+    Zvo9GI2e2MaZyo9/lvb+LbLEJZKEQckqRj4P26gmASrZEPStwc+yqy1ShHLA0j6m\
+    1QIDAQAB\
+    -----END PUBLIC KEY-----" | sed 's/   */\n/g' > "/etc/apk/keys/sgerrand.rsa.pub"
+
+# Update existing packages, and install build dependencies
+RUN echo "http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
+RUN apk upgrade --update
+RUN apk add --virtual .build-deps $BUILD_DEPS
```

Comment:
You can probably bake this in one `RUN`; I don't see the benefit from a caching pov to separate them